lib/charms/spark.py

```diff
@@ -84,6 +84,10 @@ def setup_spark_config(self):
         utils.re_edit_in_place(spark_log4j, {
             r'log4j.rootCategory=INFO, console': 'log4j.rootCategory=ERROR, console',
         })
+        dist = get_dist_config()
+        livy = Livy(dist)
+        if livy.verify_resources():
```

Comment:
Shouldn't this `if` be in `reactive/spark.py:install_spark:46` to ensure that we retry it if failed and then the `resources_mirror` URL is changed?